src/query.rs

```diff
@@ -202,32 +201,26 @@ mod tests {
             .expect("not a Vec<Vec<&str>>");
         actual_paths.sort_unstable();
 
-        let expected_paths = vec![
-            vec!["test_crates", "CheckPubUseHandling"],
-            vec!["test_crates", "import_handling", "CheckPubUseHandling"],
-            vec![
-                "test_crates",
-                "import_handling",
-                "inner",
-                "CheckPubUseHandling",
-            ],
-        ];
+        let expected_paths = vec![vec!["pub_use_handling", "CheckPubUseHandling"]];
         assert_eq!(expected_paths, actual_paths);
     }
 
-    fn check_query_execution(query_name: &str) {
-        // Ensure the rustdocs JSON outputs have been regenerated.
-        let baseline_crate = load_rustdoc(Path::new("./localdata/test_data/baseline.json"))
-            .with_context(|| "Could not load localdata/test_data/baseline.json file, did you forget to run ./scripts/regenerate_test_rustdocs.sh ?")
-            .expect("failed to load baseline rustdoc");
-        let current_crate =
-            load_rustdoc(Path::new(&format!("./localdata/test_data/{}.json", query_name)))
-            .with_context(|| format!("Could not load localdata/test_data/{}.json file, did you forget to run ./scripts/regenerate_test_rustdocs.sh ?", query_name))
-            .expect("failed to load rustdoc under test");
-
-        let baseline = VersionedIndexedCrate::new(&baseline_crate);
-        let current = VersionedIndexedCrate::new(&current_crate);
+    fn get_test_crate_names<'a>() -> Vec<String> {
+        std::fs::read_dir("./test_crates/")
+            .unwrap()
+            .map(|dir_entry| dir_entry.unwrap())
+            .filter(|dir_entry| dir_entry.metadata().unwrap().is_dir())
```

Comment:
Can you use `.expect()` instead of `.unwrap()` here with a suitable error message, to make it a bit more obvious which unwrapping failed if we get an error from one of them?